greenriver/hmis-warehouse
# 7038 user edit history

## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
[GH Issue](https://github.com/open-path/Green-River/issues/7038)

User edit history page
* fix misalignment with paginated values count and rows shown
* fix performance problems around n+1 and trying to display 500 items on one page
* ensure changes made to the Hmis::User are also shown
* show all changes on the user except for login events
* attempt to summarize changes to be better understood by end-users (invitation sent, etc)
* ensure credentials are clearly redacted in the version history
* displays impersonating users
* add tests

## Testing
As and admin , visit warehouse admin users page, select 'audit' user, select 'edit history' tab.

* login events or failed log in attempts should not be displayed
* change name, email, etc shows detailed changes
* user changes own password is summarized
* user requests a password reset  is summarized
* user changes password from reset form  is summarized
* invitation is sent  is summarized
* account is deactivated
* account is activated
* account is created

## Type of change
Bug fix, New feature, Code clean-up

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
